### ISSUE TITLE
Style/AccessorGrouping false positives with sorbet-runtime type signatures

### ISSUE THREAD

**Issue by dduugg** [rubocop/rubocop#11596]
## Expected behavior

Sorbet allows type signatures to be [attached](https://sorbet.org/docs/faq#how-should-i-add-types-to-methods-defined-with-attr_reader) to `attr_*` methods:

```ruby
class C
  extend T::Sig

  sig { returns(String) }
  attr_reader :str

  sig { returns(Integer) }
  attr_reader :int
end
```

I expect this code to be accepted by the `Style/AccessorGrouping`, as the conflicting type signatures do not allow the `attr_reader`s to be grouped.

## Actual behavior

The above code triggers `Style/AccessorGrouping` violations:

```
rubocop foo.rb --only Style/AccessorGrouping
Inspecting 1 file
C

Offenses:

foo.rb:8:3: C: [Correctable] Style/AccessorGrouping: Group together all attr_reader attributes.
  attr_reader :str
  ^^^^^^^^^^^^^^^^
foo.rb:11:3: C: [Correctable] Style/AccessorGrouping: Group together all attr_reader attributes.
  attr_reader :int
  ^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

Create a file `foo.rb` with the example at the top, then invoke `rubocop foo.rb --only Style/AccessorGrouping`

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ rubocop -V
1.45.1 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.1.1) [x86_64-darwin21]
```

**Comment by koic** [comment]
I've opened #11602. That can respect the user's intentions (e.g. annotation method) , not just Sorbet I think.
